### PR TITLE
feat(array-utils): added index param on ItemPredicate

### PR DESCRIPTION
**Pull request by NathanAlcantara**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

I cannot removing an item of collection by index in easy way, i developed a way to do that by my own:
```
propertyArrayRemoveByIndex(prop: string, index: number): void {
	let iterable = -1;
	this.store.update(state => ({
		[prop]: arrayRemove(state[prop], () => {
			iterable++;
			if (iterable === index) {
				return true;
			}
		})
	}));
}
```

but in your code i found a way to make easier that process

## What is the new behavior?

Basically you can remove an item or items of a collection using an index or indexes as a reference, like Array.prototype.splice if you pass the index as a start point and removing only 1 element (eg.: ['a', 'b', 'c'].splice(1, 1) // returns ['a', 'c'])

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by NetanelBasal** [comment]
You can use the index:

```ts
arrayRemove([1, 2, 3], (_, i) => i === 0);
```

We just need to fix the `ItemPredicate` type and add the `index` params.

**Comment by NathanAlcantara** [comment]
@NetanelBasal 
Thanks for this advice and srry for the overengineering haha
I fixed it and add some unit test
